File: fastApiProject/entity/KnowledgeChunk.py
from pydantic import BaseModel, Field
from typing import Optional, List, Dict, Any
from datetime import datetime


class KnowledgeChunk(BaseModel):
    """知识数据表（向量存储）"""
    id: Optional[int] = Field(None, description="分块唯一ID")
    knowledge_base_id: int = Field(..., description="所属知识库ID")
    knowledge_file_id: Optional[int] = Field(None, description="来源文件ID")
    chunk_index: int = Field(..., description="分块序号")
    content: str = Field(..., description="文本内容")
    embedding: Optional[List[float]] = Field(None, description="向量数据（维度与模型一致）")
    chunk_metadata: Dict[str, Any] = Field(default_factory=dict, description="元数据")
    indexing_method: Optional[str] = Field(None, max_length=50, description="建库方式")
    created_at: Optional[datetime] = Field(None, description="创建时间")
    updated_at: Optional[datetime] = Field(None, description="更新时间")
    is_deleted: int = Field(0, description="逻辑删除标志:0-未删除,1-已删除")

    class Config:
        from_attributes = True

File: blog-agent/agent/src/main/java/com/xiafan/agent/entity/KnowledgeChunk.java
package com.xiafan.agent.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.xiafan.agent.repository.PostgresJsonTypeHandler;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/** knowledge_chunks */
@TableName(value = "knowledge_chunks", autoResultMap = true)
@Data
public class KnowledgeChunk {
    @TableId(value = "id", type = IdType.AUTO)
    private Integer id;
    @TableField("knowledge_base_id")
    private Integer knowledgeBaseId;
    @TableField("knowledge_file_id")
    private Integer knowledgeFileId;
    @TableField("chunk_index")
    private Integer chunkIndex;
    @TableField("content")
    private String content;
    @TableField(value = "embedding", typeHandler = PostgresJsonTypeHandler.class)
    private List<Double> embedding;
    @TableField(value = "chunk_metadata", typeHandler = PostgresJsonTypeHandler.class)
    private Map<String, Object> chunkMetadata = new HashMap<>();
    @TableField("indexing_method")
    private String indexingMethod = "semantic";
    @TableField("created_at")
    private LocalDateTime createdAt;
    @TableField("updated_at")
    private LocalDateTime updatedAt;
    @TableField("is_deleted")
    private Integer isDeleted = 0;
}
